src/action_management/include/actionmanagement.h

```diff
@@ -0,0 +1,339 @@
+#ifndef _ACTIONS_H_
+#define _ACTIONS_H_ 
+
+#include <stdio.h>
+
+
+/* An enumeration of all supported actions. */
+/* All synonyms will be provided to CLI. */
+enum actions {
+  OPEN, /* world action */
+  CLOSE, /* world action */
+  PUSH, /* world action */
+  PULL, /* world action */
+  EXAMINE, /* synonymous to "look at", is a world action */
+  TURN_ON, /* world action */
+  TURN_OFF, /* world action */
+  TALK_TO, /* NPC action */
+  GIVE, /* requires 2 arguments,  NPC action */
+  GO, /* world action */
+  TAKE, /* synonymous with "pick up", is an inventory action */
+  DROP, /* inventory action */
+  CONSUME, /* synonymous with "use", "drink", "eat", is an inventory action */
+  USE_ON, /* world and NPC action */
+  MEDLY /* NPC action */
+};
+
+
+/* An enumeration of all possible action types. */
+enum action_type {
+  ACT_NPC, /* includes "give", "talk to" */
+  ACT_WORLD, /* includes "look at", "examine", "open", "close", "push", "pull", 
+		"turn on", "turn off", and "go" */
+  ACT_INVENTORY /* includes "take", "drop", "consume" */
+};
+  
+
+/* An action struct that contains the following: 
+ * act: A tag corresponding to the action 
+ * type: A tag describing the type of action
+ * num_arg: the number of actions required for the action to be valid
+ * success: a string indicating the action was a success
+ * failure: a string indicating the action was a failure
+*/
+struct action_struct {
+  enum actions act;
+  enum action_type type;
+  int num_arg; 
+  char *success;
+  char *failure;
+};
+
+
+/* A type declaration of the action struct to action_t. */
+typedef struct action_struct action_t;
+
+
+/* 
+ * Allocates a new action on the heap
```

Comment:
In reference to the next comment:

```suggestion
 * Allocates a new action on the heap, calling action_init
```